DROD/SettingsScreen.cpp

```diff
@@ -1052,6 +1109,86 @@ void CSettingsScreen::OnClick(const UINT dwTagNo)
 #  undef BUTTONTAG_TO_DCMD
 }
 
+//************************************************************************************
+const KeyDefinition *CSettingsScreen::GetOverwrittenModifiableKey(const InputKey newKey, const DCMD eChangedCommand) const
+{
+	const SDL_Keycode newKeycode = ReadInputKey(newKey);
+
+	for (int nCommand = DCMD_NW; nCommand < DCMD_Count; ++nCommand)
+	{
+		if (nCommand == eChangedCommand || !DoesCommandUseModifiers((DCMD)nCommand))
+			continue;
+
+		const KeyDefinition *keyDefinition = GetKeyDefinition(nCommand);
+
+		if (this->pCurrentPlayer->Settings.GetVar(keyDefinition->settingName, UNKNOWN_INPUT_KEY) == (InputKey)newKeycode)
+			return keyDefinition;
+	}
+
+	return NULL;
+}
+
+//************************************************************************************
+void CSettingsScreen::DoKeyRedefinition(const UINT dwTagNo) {
+	if (dwTagNo < TAG_KEY_BUTTON_START || dwTagNo > TAG_KEY_BUTTON_END)
+		return;
+
+	InputKey newKey, currentKey;
+
+	const DCMD eCommand = ButtonTagToDcmd(dwTagNo);
+	const KeyDefinition *keyDefinition = GetKeyDefinition(eCommand);
+	currentKey = this->pCurrentPlayer->Settings.GetVar(keyDefinition->settingName, UNKNOWN_INPUT_KEY);
+
+	if (GetCommandKeyRedefinition(eCommand, currentKey, newKey, !DoesCommandUseModifiers(eCommand)))
```

Comment:
For readability, suggest inverting the check here (i.e, prepend "!") and place the else block immediately underneath it.  Then you don't need all the rest of the code indented inside another block below that.